src/main/java/com/squad9/bluebank/service/EnderecoServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.squad9.bluebank.service;
+
+import com.squad9.bluebank.dto.EnderecoRequestDTO;
+import com.squad9.bluebank.dto.EnderecoResponseDTO;
+import com.squad9.bluebank.model.Cliente;
+import com.squad9.bluebank.model.Endereco;
+import com.squad9.bluebank.repository.EnderecoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class EnderecoServiceImpl implements EnderecoService{
+
+    @Autowired
+    private EnderecoRepository enderecoRepository;
+
+
+    //Cadastrar novo endereço.
+    public EnderecoResponseDTO salvarEndereco(EnderecoRequestDTO enderecoRequestDTO, Cliente cliente) { // Coloquei "Cliente cliente" porque precisa indentificar o endereço do cliente.
```

Comment:
Aqui no lugar de :
`Cliente cliente`
Não seria ?
`Long idCliente`

Eu me baseie no que foi feito no método atualizarEndereço, onde recebe o id do endereço e não uma instancia de model endereço diretamente, sacou?

Ai, se for receber o idCliente, poderia colocar depois a mesma verificação que você fez para o id do endereço dentro do método atualizarEndereco para verificar se o id é respectivo a um endereço existente. No caso do cliente ficaria:

`var cliente = this.clienteRepository.findById(idCliente).orElseThrow(()->new Exception("Cliente não encontrado"));`

Reply:
Certo, alterei já